api/secrets.proto

```diff
@@ -17,13 +17,13 @@ syntax = "proto3";
 package ttn.lorawan.v3;
 
 import "github.com/gogo/protobuf/gogoproto/gogo.proto";
-import "github.com/envoyproxy/protoc-gen-validate/validate/validate.proto";
 
 option go_package = "go.thethings.network/lorawan-stack/v3/pkg/ttnpb";
 
-// Secrets contains key-value pairs of secret values as well as the ID of the encryption key.
-message Secrets{
-  map<string,bytes> values = 1 [(validate.rules).map.keys.string = {pattern: "^[a-z0-9](?:[-]?[a-z0-9]){2,}$" , max_len: 36}];
+// Secret contains a secret blob as well as the ID of the encryption key.
+// The blob is typically a marshalled proto message.
+message Secret{
```

Comment:
So how does this relate to the gateway secrets?

Reply:
So this `Secret` contains an encrypted blob with the key ID primarily meant for storing. 

The `GatewaySecrets` is how plaintext is handled by clients and is meant for RPCs.